runtime/errors.go

```diff
@@ -20,6 +21,17 @@ type StreamErrorHandlerFunc func(context.Context, error) *status.Status
 // RoutingErrorHandlerFunc is the signature used to configure error handling for routing errors.
 type RoutingErrorHandlerFunc func(context.Context, *ServeMux, Marshaler, http.ResponseWriter, *http.Request, int)
 
+// StatusHTTPError is the error to use when needing to provide a different status code for an error
+// passed to the DefaultRoutingErrorHandler
+type StatusHTTPError struct {
+	Status int
```

Comment:
Done, I think, did you just mean the leading part to keep the trailing `Error` or rename the full, I realized after updating that it could mean either?

Reply:
Haha, sorry for the confusion, I _did_ mean the field, and I do think we should keep the `Error` suffix in this type. How about we call the type `HTTPError` and call the field `HTTPStatus`?

Reply:
As there is a function in the packa called `HTTPError`, it appears that causes confusion for the compiler as how to distinguish between them. Switched it to HTTPStatusError for now.